pyvista/plotting/renderer.py

```diff
@@ -1267,6 +1276,17 @@ def show_bounds(
             raise ValueError(f'padding ({padding}) not understood. Must be float between 0 and 1')
         cube_axes_actor.SetBounds(bounds)
 
+        # update the values of range of each axis
+        if isinstance(axes_range, (collections.abc.Sequence, np.ndarray)):
+            if len(axes_range) == 6:
+                cube_axes_actor.SetXAxisRange(axes_range[0], axes_range[1])
+                cube_axes_actor.SetYAxisRange(axes_range[2], axes_range[3])
+                cube_axes_actor.SetZAxisRange(axes_range[4], axes_range[5])
+            else:
+                raise ValueError(
+                    'axes_range must be passed as a [xmin, xmax, ymin, ymax, zmin, zmax] list or tuple'
+                )
```

Comment:
We should probably raise here `if axes_range is not None`, telling the user to do better.

Reply:
Github is being weird. In the "files" tab you can see the predecessor to this comment.

Reply:
Where is the exception raised when we pass in `axes_ranges = 'potato'`?

Reply:
I see what you mean. I was assuming that the user wouldn't be passing anything absurd into the function.

Is this more agreeable?

```python

        # update the range values of each axis
        if axes_ranges is None:
            axes_ranges = bounds
        elif isinstance(axes_ranges, (collections.abc.Sequence, np.ndarray)):
            # Make sure the elements are all compatible 
            if not all(isinstance(i , (int, float)) for i in axes_ranges):
                raise TypeError('each element in axes ranges must be passed as an int or float')
            
            if len(axes_ranges) == 6:
                cube_axes_actor.SetXAxisRange(axes_ranges[0], axes_ranges[1])
                cube_axes_actor.SetYAxisRange(axes_ranges[2], axes_ranges[3])
                cube_axes_actor.SetZAxisRange(axes_ranges[4], axes_ranges[5])
            else:
                raise ValueError(
                    'axes_ranges must be passed as an [xmin, xmax, ymin, ymax, zmin, zmax] list or tuple'
                )
```

Reply:
Not really, because this tackles the parentheses in my previous comment for which I said "let's ignore that for now" :P Not ignoring it is fine, but it's not what I had in mind originally. I'm talking about an `else` clause here!

As for the above snippet: it won't work for numpy array input. Instead I'd recommend converting `axes_ranges` to an array inside that `elif`, and checking the type. This is a bit fiddly which is why I suggested to ignore it for now. It's a larger issue if someone passes garbage input. "user passing something absurd into the function" is exactly that input validation should prepare for.

Reply:
I interpreted your "let's ignore that for now" as, "let's ignore the fact that strings match the `is instance`". 

Honestly - this seems pedantic to me here... It seems excessive to write an error for every possible instance of potential argument that the user might pass. For example, if you pass in `plotter.show_bounds(bounds='potato')`, you only get a `numpy` error. 

Below fixes the np.array not passing correctly issue.
```python
if axes_ranges is None:
    axes_ranges = bounds
elif isinstance(axes_ranges, (collections.abc.Sequence, np.ndarray)):
    axes_ranges = np.asarray(axes_ranges)
    # make sure the elements are all compatible 
    if not isinstance(axes_ranges.dtype, (np.int_, np.float_)):
        raise TypeError('each element in axes_ranges must be passed as an int or float')

    if len(axes_ranges) == 6:
        cube_axes_actor.SetXAxisRange(axes_ranges[0], axes_ranges[1])
        cube_axes_actor.SetYAxisRange(axes_ranges[2], axes_ranges[3])
        cube_axes_actor.SetZAxisRange(axes_ranges[4], axes_ranges[5])
    else:
        raise ValueError(
            'axes_ranges must be passed as an [xmin, xmax, ymin, ymax, zmin, zmax] list or tuple'
        )
```

If not this, how exactly are you hoping for me to catch non-confirming inputs? Thanks

Reply:
Or this if not the former? This won't give a meaningful error to `axes_ranges=[0,1,0,1,0,'b']`, but seems a bit cleaner.

```python
if axes_ranges is None:
    axes_ranges = bounds
elif isinstance(axes_ranges, (list, tuple, np.ndarray)):
    if len(axes_ranges) == 6:
        cube_axes_actor.SetXAxisRange(axes_ranges[0], axes_ranges[1])
        cube_axes_actor.SetYAxisRange(axes_ranges[2], axes_ranges[3])
        cube_axes_actor.SetZAxisRange(axes_ranges[4], axes_ranges[5])
    else:
        raise ValueError(
            'axes_ranges must be passed as an [xmin, xmax, ymin, ymax, zmin, zmax] list or tuple'
        )
else:
    raise TypeError('axes_ranges must be passed as a list, tuple, or np.ndarray with int or float elements')
```

Reply:
Yes, it's pedantic; I thought we'd met before :smile: But it's not pointless pedantry. Let me explain my stance.

If you do
```py
if cond1:
    foo  # handle default
elif cond2:
    bar  # handle sane inputs
baz  # rest of the code
```
you make the decision to silently ignore garbage input. That's not what Python usually does, and that's arguably not what a library (having downstream libraries as well as lots of users) should do. What I recommend is your last snippet:
```py
if cond1:
    foo  # handle default
elif cond2:
    bar  # handle sane inputs
else:
    raise TypeError("I don't think so.")
```

Of course nobody in their right mind would pass `{'potato'}`. If someone intentionally passes garbage they had it coming. But what _might_ happen is that someone passes the wrong variable by accident. Say, instead of passing `other_mesh.bounds` they pass `other_mesh`, which is not a sequence. In this case we would just silently ignore the `axes_ranges` kwarg, even though someone went out of their way to specify it.

Handling almost-valid inputs like `axes_ranges=[0,1,0,1,0,'b']` is not something we should always prepare for, at least not with too helpful error messages. This is a subtler issue, but an error elsewhere is an easier thing to track down than a silent no-op. We _could_ handle this case too with my earlier recommendation, but we don't really _have_ to; this would be something like
```py
elif isinstance(axes_ranges, (collections.abc.Sequence, np.ndarray):
    axes_ranges = np.asanyarray(axes_ranges)
    if not np.issubdtype(axes_ranges.dtype, np.number):
        raise TypeError(f'Input axes_ranges must be a numeric sequence.')
```
This would work because if there's a single string in there then the dtype would be `object`. And after this point we can just check that the shape of the range is `(6,)`. Again: we don't have to do this. I only feel strongly about my original point.

Reply:
>Yes, it's pedantic; I thought we'd met before 😄 But it's not pointless pedantry. Let me explain my stance.

🤣 touché

>Of course nobody in their right mind would pass {'potato'}. If someone intentionally passes garbage they had it coming. But what might happen is that someone passes the wrong variable by accident. Say, instead of passing other_mesh.bounds they pass other_mesh, which is not a sequence. In this case we would just silently ignore the axes_ranges kwarg, even though someone went out of their way to specify it.

This scenario makes much more sense to me. I suppose I wasn't being creative enough to visualize it. :)

Based on your input, I'll go ahead and propose the following to try to deal with egregiously incorrect inputs as well as inputs that the user might not be aware of passing incorrectly...

```python
if axes_ranges is None:
    axes_ranges = bounds
elif isinstance(axes_ranges, (collections.abc.Sequence, np.ndarray)):
    axes_ranges = np.asarray(axes_ranges)
    if not isinstance(axes_ranges, np.number):
        raise TypeError(f'input axes_ranges must be a numeric sequence.')
    
    if axes_ranges.shape[0]:
        cube_axes_actor.SetXAxisRange(axes_ranges[0], axes_ranges[1])
        cube_axes_actor.SetYAxisRange(axes_ranges[2], axes_ranges[3])
        cube_axes_actor.SetZAxisRange(axes_ranges[4], axes_ranges[5])
    else:
        raise ValueError(
            'axes_ranges must be passed as an [xmin, xmax, ymin, ymax, zmin, zmax] list or tuple'
        )
else:
    raise TypeError('input axes_ranges must be a numeric sequence')

```

Reply:
Thanks for being patient enough to explain your thought process and some more of the design philosophy. I appreciate it :)

Reply:
> Thanks for being patient 

And the same to you ;)

> Based on your input, I'll go ahead and propose the following [...]

There are a few things that need editing in there:
1. `np.asanyarray` (note the "any") on line 4
2. take a closer look at my last snippet in my previous comment: `isinstance` -> `issubdtype` and using `axes_ranges.dtype`. Since we'll always have a numpy ndarray here, what we have to do is confirm that it has numbers inside.
3. `if axes_ranges.shape[0]` should be `if axes_range.shape == (6,)` to make sure we don't have something weird like a 2d array. (Or we could `ravel()` the array before working with it...)

You can also just go ahead and make the change and we can iron out the kinks as suggested changes on the PR.

Reply:
One style typo that I'll fix in a moment, otherwise the current version looks good to me, thanks.